src/libraries/System.Private.CoreLib/src/System/Guid.Unix.cs

```diff
@@ -9,7 +11,10 @@ public partial struct Guid
         public static unsafe Guid NewGuid()
         {
             Guid g;
-            Interop.GetRandomBytes((byte*)&g, sizeof(Guid));
+
+            int res = Interop.Sys.GetCryptographicallySecureRandomBytes((byte*)&g, sizeof(Guid));
```

Comment:
Can we add a comment about why we're doing this? I'd like to avoid a perf investigation in a couple of years changing it back because "cryptographically-secure isn't required here". ;-)